pkg/splunk/enterprise/indexercluster.go

```diff
@@ -338,38 +287,21 @@ func getIndexerStatefulSet(client splcommon.ControllerClient, cr *enterprisev1.I
 	return getSplunkStatefulSet(client, cr, &cr.Spec.CommonSplunkSpec, SplunkIndexer, cr.Spec.Replicas, getIndexerExtraEnv(cr, cr.Spec.Replicas))
 }
 
-// getClusterMasterStatefulSet returns a Kubernetes StatefulSet object for a Splunk Enterprise license master.
-func getClusterMasterStatefulSet(client splcommon.ControllerClient, cr *enterprisev1.IndexerCluster) (*appsv1.StatefulSet, error) {
-	return getSplunkStatefulSet(client, cr, &cr.Spec.CommonSplunkSpec, SplunkClusterMaster, 1, getIndexerExtraEnv(cr, cr.Spec.Replicas))
-}
-
 // validateIndexerClusterSpec checks validity and makes default updates to a IndexerClusterSpec, and returns error if something is wrong.
-
 func validateIndexerClusterSpec(cr *enterprisev1.IndexerCluster) error {
-	// IndexerCluster must support 0 replicas for multisite / multipart clusters
-	// Multisite / multipart clusters: can't reference a cluster master located in another namespace because of Service and Secret limitations
-	if len(cr.Spec.IndexerClusterRef.Namespace) > 0 && cr.Spec.IndexerClusterRef.Namespace != cr.GetNamespace() {
-		return fmt.Errorf("Multisite cluster does not support cluster master to be located in a different namespace")
+	// We cannot have 0 replicas in IndexerCluster spec, since this refers to number of indexers in an indexer cluster
+	if cr.Spec.Replicas == 0 {
+		cr.Spec.Replicas = 1
 	}
 
-	if cr.Spec.IndexerClusterRef.Name == "" {
-		err := ValidateSplunkSmartstoreSpec(&cr.Spec.SmartStore)
-		if err != nil {
-			return err
-		}
-	} else if isSmartstoreConfigured(&cr.Spec.SmartStore) {
-		return (fmt.Errorf("Smartstore configuration is only supported with Cluster Master spec"))
+	// Cannot leave clusterMasterRef field empty or else we cannot connect to CM
+	if len(cr.Spec.ClusterMasterRef.Name) == 0 {
+		return fmt.Errorf("IndexerCluster spec should refer to ClusterMaster via clusterMasterRef")
 	}
 
-	return validateCommonSplunkSpec(&cr.Spec.CommonSplunkSpec)
-}
-
-// getIndexerExtraEnv returns extra environment variables used by search head clusters
-func getIndexerExtraEnv(cr splcommon.MetaObject, replicas int32) []corev1.EnvVar {
-	return []corev1.EnvVar{
-		{
-			Name:  "SPLUNK_INDEXER_URL",
-			Value: GetSplunkStatefulsetUrls(cr.GetNamespace(), SplunkIndexer, cr.GetName(), replicas, false),
-		},
+	// Multisite / multipart clusters: can't reference a cluster master located in another namespace because of Service and Secret limitations
+	if len(cr.Spec.ClusterMasterRef.Namespace) > 0 && cr.Spec.ClusterMasterRef.Namespace != cr.GetNamespace() {
+		return fmt.Errorf("Multisite cluster does not support cluster master to be located in a different namespace")
```

Comment:
AFAIK, we are filing just the name, and other fields are empty? Are we filling namespace somewhere?

Reply:
You can if you want to, but we are not supporting that. That's why this check :)

Reply:
I see.. i was thinking it as a string:-)
IndexerClusterRef corev1.ObjectReference `json:"indexerClusterRef"`